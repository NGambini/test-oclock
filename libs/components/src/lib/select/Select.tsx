import React, { FC, useState } from 'react';
import MaterialIcon from '@material/react-material-icon';
import { animated, useSpring } from 'react-spring';
import { usePopper } from 'react-popper';
import classNames from 'classnames';

import styles from './select.module.less';

export type SelectProps = {
  placeholder?: string;
};

export const Select: FC<SelectProps> = ({ placeholder }) => {
  const [isOpened, setIsOpened] = useState(false);
  const iconRotationAngle = useSpring({
    transform: isOpened ? 'rotate(180deg)' : 'rotate(0deg)',
  });
  const [referenceElement, setReferenceElement] = useState(null);
  const [popperElement, setPopperElement] = useState(null);
  const { styles, attributes } = usePopper(referenceElement, popperElement);

  const handleOnClick = () => {
    setIsOpened(!isOpened);
  };

  return (
    <div>
      <div
        ref={setReferenceElement}
        className={classNames(styles['container'])}
        onClick={handleOnClick}
      >
        {placeholder}

        <animated.div
          className={classNames(styles['iconContainer'])}
          style={iconRotationAngle}
        >
          <MaterialIcon icon="expand_more" />
        </animated.div>
      </div>
      {isOpened && (
        <div
          ref={setPopperElement}
          style={styles.popper}
          {...attributes.popper}
        >
          le popper <br />
          le popper <br />
          le popper <br />
          le popper <br />
          le popper <br />
          le popper <br />
          le popper <br />
          le popper <br />
        </div>
      )}
    </div>
  );
};
