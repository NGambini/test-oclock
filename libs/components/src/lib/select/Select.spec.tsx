import React from 'react';
import { render } from '@testing-library/react';

import { Select } from './Select';

describe('Select', () => {
  it('renders correctly', () => {
    const { baseElement: select } = render(
      <Select
        icon="language"
        placeholder={'Convert to...'}
        value="1"
        options={[
          { key: '1', name: 'First' },
          { key: '2', name: 'Second' },
          { key: '3', name: 'Third' },
        ]}
      />
    );
    expect(select).toMatchSnapshot();
  });
});
