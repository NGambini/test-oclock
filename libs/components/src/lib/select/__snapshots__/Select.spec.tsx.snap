// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<body>
  <div>
    <div
      data-cy="select"
    >
      <div
        class=""
      >
        <div
          class="rippleContainer"
        />
        <i
          class="material-icons"
        >
          language
        </i>
        <div>
          First
        </div>
        <div
          style="transform: rotate(0deg);"
        >
          <i
            class="material-icons"
          >
            expand_more
          </i>
        </div>
      </div>
      <div
        class=""
        style="position: absolute; left: 0px; top: 0px;"
      >
        <div>
          <div>
            <div
              class="undefined"
              data-cy="option"
            >
              <i
                class="material-icons"
              >
                check
              </i>
              <span>
                First
              </span>
            </div>
            <div
              class=""
              data-cy="option"
            >
              <span>
                Second
              </span>
            </div>
            <div
              class=""
              data-cy="option"
            >
              <span>
                Third
              </span>
            </div>
          </div>
          <div />
        </div>
      </div>
    </div>
  </div>
</body>
`;
